### DISCUSSION TITLE
kafka without zookeeper

### DISCUSSION THREAD

**Discussion by MK-BK** [strimzi/strimzi-kafka-operator#6733] [Q&A]
When will kafka kraft be supported

**Comment by scholzj** [comment]
It is not done and ready in Apache Kafka it self yet and there is no clear date there yet either. So I don't think we can give you any date.

**Reply by MK-BK** [reply]
How to understand the TODO task of zookeeper-less kafka

**Reply by scholzj** [reply]
What TODO task do you mean?

**Reply by MK-BK** [reply]
![image](https://user-images.githubusercontent.com/43365367/165456740-00f59564-b9e7-45d4-a47c-05258cdab92a.png)

**Reply by scholzj** [reply]
Ah, ok. Well, we know it is coming. We are already working on some preparation tasks, doing some prototyping etc. But it cannot be _production ready_ in Strimzi before it is in Apache Kafka it self. So it is hard to do any planning or set some exact timelines.

**Reply by scholzj** [reply]
Or to put it differently ... we obviously want to be ready for it as soon as possible. So we are not waiting for it to be done in Apache Kafka before we start doing anything. But at the same time, maintaining many different code paths costs a lot of effort. So we also need to be careful about when it is offered for example as some feature for testing because it might significantly slow down all other work.

**Reply by MK-BK** [reply]
Ok, Thanks for your reply.
